pkg/porter/delete.go

```diff
@@ -0,0 +1,53 @@
+package porter
+
+import (
+	"fmt"
+
+	"get.porter.sh/porter/pkg/context"
+	claims "github.com/cnabio/cnab-go/claim"
+	"github.com/pkg/errors"
+)
+
+const installationDeleteTmpl = "deleting installation records for %s...\n"
+
+// DeleteOptions represent options for Porter's installation delete command
+type DeleteOptions struct {
+	sharedOptions
+	Force bool
+}
+
+// Validate prepares for an installation delete action and validates the args/options.
+func (o *DeleteOptions) Validate(args []string, cxt *context.Context) error {
+	// Ensure only one argument exists (installation name) if args length non-zero
+	err := o.sharedOptions.validateInstallationName(args)
+	if err != nil {
+		return err
+	}
+
+	return o.sharedOptions.defaultBundleFiles(cxt)
+}
+
+// DeleteInstallation handles deletion of an installation
+func (p *Porter) DeleteInstallation(opts DeleteOptions) error {
+	err := p.applyDefaultOptions(&opts.sharedOptions)
+	if err != nil {
+		return err
+	}
+
+	// TODO: do we check to see if the installation claim has required deps
```

Comment:
I believe all the logic for dependencies should be encapsulated in the actions, e.g. uninstall. So if they have made it to this point, either they are forcing the deletion or it was uninstalled and the deps were taken care of already.